### PR TITLE
[FIX] hr_expense: expense{_sheet} state reset move to draft

### PR DESCRIPTION
**Pull request by guva-odoo**
Override the button_draft method, in order to
reset the expense and expense_sheet state
when resetting a move to draft.

Steps:

- Create an expense
- Submit and approve the report
- Post journal entry and register payment
- Reset the JE to draft
-> The expense and expense sheet states
   remain the same as before resetting
   to draft. They should be respectively
   'approved' and 'post'

opw-3279364

### CODE REVIEW

**Comment by guva-odoo** [comment]
@fw-bot up to saas-16.1

File: addons/hr_expense/models/account_move.py [hunk 1]
```diff
@@ -12,3 +12,8 @@ def button_cancel(self):
             if l.expense_id:
                 l.expense_id.refuse_expense(reason=_("Payment Cancelled"))
         return super().button_cancel()
+
+    def button_draft(self):
+        for line in self.line_ids:
+            line.expense_id.sheet_id.write({'state': 'post'})
```
Comment (epictete) [inline addons/hr_expense/models/account_move.py:18]: I would add a condition to check that the line has an `expense_id`, like above in `button_cancel()`

File: addons/hr_expense/tests/test_expenses.py [hunk 2]
```diff
@@ -350,3 +350,63 @@ def test_print_expense_check(self):
...
+        self.assertEqual(expense_sheet.state, 'approve', 'Sheet state must be draft after approval')
+
+        # Create move
+        res = expense_sheet.action_sheet_move_create()
+
+        self.assertEqual(expense.state, 'approved', 'Expense state must be draft after posting move')
+        self.assertEqual(expense_sheet.state, 'post', 'Sheet state must be draft after posting move')
+
+        # Pay move
+        move = self.env['account.move'].browse(res[expense_sheet.id].id)
```
Comment (epictete) [inline addons/hr_expense/tests/test_expenses.py:391]: Why browse `account.move` when you already have the record with `res[expense_sheet.id]`?

File: addons/hr_expense/tests/test_expenses.py [hunk 3]
```diff
@@ -350,3 +350,63 @@ def test_print_expense_check(self):
...
+        self.env['account.payment.register'].with_context(active_model='account.move', active_ids=move.ids).create({
+            'amount': 1000.0,
+        })._create_payments()
+
+        self.assertEqual(expense.state, 'done', 'Expense state must be done after payment')
+        self.assertEqual(expense_sheet.state, 'done', 'Sheet state must be done after payment')
+
+        # Reset move to draft
+        self.env['account.move'].browse(move.id).button_draft()
```
Comment (epictete) [inline addons/hr_expense/tests/test_expenses.py:400]: Do you see something weird/useless here? :wink:

Comment (guva-odoo) [reply addons/hr_expense/tests/test_expenses.py:400]: ahahahahah, end of the day...

Comment (guva-odoo) [reply addons/hr_expense/tests/test_expenses.py:400]: I tried `self.env['account.move'].browse(self.env['account.move'].browse(move.id).id)` but I fount it a little overkill

File: addons/hr_expense/tests/test_expenses.py (hunk 2 above)
Comment (guva-odoo) [reply addons/hr_expense/tests/test_expenses.py:391]: I could do like here https://github.com/odoo/odoo/blob/5c74a71b53b83630a8d4eb1187aacb3d7fd92b91/addons/hr_expense/tests/test_expenses.py#L292-L294
but as I need the move later, I believe it is ok to browse it here and pay it directly, what do you think?

**Comment by epictete** [comment]
@robodoo r+

**Pull request closed without merging**